File: InsertionSort.cpp
#include<iostream>
using namespace std;

class InsertionSort {

	public:

	void sort(int array[], int length){
		
		int temp;
		
		for(int i=0; i<=length-1; i++){
			for(int j=i; j>=0; j--){
				
				if(array[j-1] > array[j]){
					temp = array[j];
					array[j] = array[j-1];
					array[j-1] = temp;
				}
			}
		}
			
		for(int i=0; i<length; i++){
			cout << array[i] << "\t";
		}
	}
};


int main(){
	
	int array[] = {98, 1, 45, 38, 22};
	int length = sizeof(array) / sizeof(array[0]);
	
	InsertionSort obj;
	obj.sort(array, length);
	
	return 0;
}

File: InsertionSort.java
import java.util.Scanner;

public class InsertionSort {
	
	public void sort(int arr[]){
		for(int i=0; i<arr.length; i++){
			
			int j = i;
			while(j>0){
				if(arr[j] < arr[j-1]){
					int temp = arr[j];
					arr[j] = arr[j-1];
					arr[j-1] = temp;
				}	
				j--;
			}
			printArray(arr);
		}
	}
	
	public void printArray(int arr[]){
		for(int i=0; i<arr.length; i++){
			System.out.print(arr[i]+" ");
		}
		System.out.println();
	}
	
	public static void main(String...strings){
		
		InsertionSort obj = new InsertionSort();
		Scanner in = new Scanner(System.in);
		
		int size = in.nextInt();
		int[] arr = new int[size];
		
		for(int i=0; i<size; i++){
			arr[i] = in.nextInt();
		}
		obj.sort(arr);
		in.close();
	}

}
